File: fix_layout.py
import re

with open('frontend/app/layout.jsx', 'r', encoding='utf-8') as f:
    layout = f.read()

# Replace metadata and add viewport
old_metadata = """export const metadata = {
  title: 'SatyaLabel',
  description: 'Legal Metrology Compliance Checker',
}"""

new_metadata = """export const metadata = {
  title: 'SatyaLabel',
  description: 'Legal Metrology Compliance Checker',
  manifest: '/manifest.json',
  appleWebApp: {
    capable: true,
    statusBarStyle: 'black-translucent',
    title: 'SatyaLabel',
  },
}

export const viewport = {
  themeColor: '#ffffff',
  width: 'device-width',
  initialScale: 1,
  maximumScale: 1,
  userScalable: false,
  viewportFit: 'cover',
};"""

layout = layout.replace(old_metadata, new_metadata)

# Remove the hardcoded head tags since Next.js metadata API handles it better
head_pattern = r'<head>.*?</head>'
layout = re.sub(head_pattern, '<head></head>', layout, flags=re.DOTALL)

with open('frontend/app/layout.jsx', 'w', encoding='utf-8') as f:
    f.write(layout)
print("Updated layout.jsx with proper PWA metadata and viewportFit: 'cover'")

File: fix_layout.js
const fs = require('fs');
let code = fs.readFileSync('frontend/app/layout.jsx', 'utf8');

// 1. Add icons to metadata
code = code.replace(
  "description: 'Legal Metrology Compliance Checker',",
  "description: 'Legal Metrology Compliance Checker',\n  icons: {\n    icon: '/icon.png',\n    shortcut: '/icon.png',\n    apple: '/icon.png',\n  },"
);

// 2. Change viewport themeColor
const oldViewportMatch = /export const viewport = \{[\s\S]*?\}/;
const newViewport = `export const viewport = {
  themeColor: '#1E3A8A',
  width: 'device-width',
  initialScale: 1,
  maximumScale: 1,
  userScalable: false,
}`;

code = code.replace(oldViewportMatch, newViewport);

fs.writeFileSync('frontend/app/layout.jsx', code);
console.log("LAYOUT RESTORED AND FIXED");
